compiler/ilgen/TypeDictionary.cpp

```diff
@@ -475,11 +592,25 @@ TypeDictionary::PointerTo(TR::IlType *baseType)
    }
 
 TR::IlReference *
-TypeDictionary::FieldReference(const char *structName, const char *fieldName)
+TypeDictionary::FieldReference(const char *typeName, const char *fieldName)
    {
    TR_HashId structID = 0;
-   _structsByName->locate(structName, structID);
-   StructType *theStruct = (StructType *) _structsByName->getData(structID);
-   return theStruct->getFieldSymRef(fieldName);
+
+   auto found = _structsByName->locate(typeName, structID);
+   if (found)
+      {
+      StructType *theStruct = (StructType *) _structsByName->getData(structID);
+      return theStruct->getFieldSymRef(fieldName);
+      }
+
+   found = _unionsByName->locate(typeName, structID);
+   if (found)
+      {
+      UnionType *theUnion = (UnionType *) _unionsByName->getData(structID);
+      return theUnion->getFieldSymRef(fieldName);
+      }
+
```

Comment:
What happens if you have a struct field and a union field with the same name, and you actually want the one from the union?